src/rulesets/Base/array.jl

```diff
@@ -102,3 +102,25 @@ function rrule(::typeof(fill), value::Any, dims::Int...)
     end
     return fill(value, dims), fill_pullback
 end
+
+#####
+##### getindex
+#####
+
+function rrule(::typeof(getindex), x::Array{<:Number}, inds::Vararg{Int})
+    y = getindex(x, inds...)
+    function getindex_pullback(ȳ)
+        function getindex_add!(Δ)
+            Δ[inds...] = Δ[inds...] .+ ȳ
```

Comment:
Out of scope for this PR probably but i wonder if the way to extend this for more things like `Array{AbstractArray}` is:

```suggestion
            Δ[inds...] = add!!(Δ[inds...], ȳ)
```